pkg/plugin/service.go

```diff
@@ -77,12 +77,12 @@ func (r *Service) ValidateSourceConfig(ctx context.Context, d Dispenser, setting
 		return cerrors.Errorf("could not dispense source: %w", err)
 	}
 
-	defer func(src SourcePlugin, ctx context.Context) {
-		err := src.Teardown(ctx)
-		if err != nil {
-			return
+	defer func() {
+		terr := src.Teardown(ctx)
+		if err == nil {
+			err = terr // only overwrite error if it's nil
```

Comment:
For this to work we need to make `err` a [named return value](https://go.dev/tour/basics/7).